src/test/java/org/example/auctions/services/AuctionServiceTest.java

```diff
@@ -0,0 +1,163 @@
+        Auction auction = auctionService.createAuction("Android", type.INC, 500, duration.H2, "Eldor");
+        auctionService.deleteAuction(auction.getId());
+        assertEquals(status.DELETED, auction.getStatus());
+    }
+
+    @Test
+    public void verifyUpdateAuction() {
+        Auction auction = auctionService.createAuction("Android", type.INC, 500, duration.H2, "Eldor");
+        auctionService.updateAuction(auction.getId(), "Iphone", type.INC, 200, duration.H2);
+        assertEquals("Iphone", auction.getSubject());
+        assertEquals(type.INC, auction.getType());
+        assertEquals(200, auction.getPrice());
+        assertEquals(duration.H2, auction.getDuration());
+    }
+
+    @Test
+    public void validateCreateAuctionThrowsWhenPriceLessZero() {
+        RuntimeException thrown = assertThrows(RuntimeException.class, () -> {
+            Auction auction = auctionService.createAuction("Android", type.INC, -1, duration.H2, "Eldor");
+        });
+        assertEquals("Price is less than zero.", thrown.getMessage());
+    }
+
+    @Test
+    public void validateUpdateAuctionThrowsExceptionWhenAuctionIdNotFound() {
+        RuntimeException thrown = assertThrows(RuntimeException.class, () -> {
+            auctionService.updateAuction("non-existent Id", "Iphone", type.INC, 200, duration.H2);
+        });
+        assertEquals("There is no such Auction Id.", thrown.getMessage());
+    }
+
+    @Test
+    public void validateUpdateAuctionThrowsExceptionWhenAuctionPublished() {
+        RuntimeException thrown = assertThrows(RuntimeException.class, () -> {
+            Auction auction = auctionService.createAuction("Android", type.INC, 500, duration.H2, "Eldor");
+            auction.publish();
+            auctionService.updateAuction(auction.getId(), "Iphone", type.INC, 200, duration.H2);
+        });
+        assertEquals("The auction is published, it cannot be updated.", thrown.getMessage());
+    }
+
+    @Test
+    public void validateUpdateAuctionThrowsExceptionWhenAuctionDeleted() {
+        RuntimeException thrown = assertThrows(RuntimeException.class, () -> {
+            Auction auction = auctionService.createAuction("Android", type.INC, 500, duration.H2, "Eldor");
+            auction.delete();
+            auctionService.updateAuction(auction.getId(), "Iphone", type.INC, 200, duration.H2);
+        });
+        assertEquals("The auction is deleted, it cannot be updated.", thrown.getMessage());
+    }
+
+    @Test
+    public void verifyCreateBid() {
+        Auction auction = auctionService.createAuction("Android", type.INC, 500, duration.H2, "Eldor");
+        Bid bid = auctionService.createBid("Shamil", auction.getId(), 600);
+        assertNotNull(bid);
+        assertEquals(1, auction.getBids().size());
+    }
+
+    @Test
+    public void validateCreateBidThrowsExceptionWhenAuctionIdNotFound() {
+        RuntimeException thrown = assertThrows(RuntimeException.class, () -> {
```

Comment:
A variable name is not the best one "thrown". Maybe "exception" at least? Or "expectedException"?